### PR TITLE
Gomod Updates

### PR DESCRIPTION
**Pull request by wild-endeavor**
This updates the boilerplate to use go modules instead of dep and follows the changes that have already been made:

* lyft/datacatalog#21
* lyft/flytepropeller#38
* lyft/flyteadmin#35
* lyft/flyteplugins#36
* lyft/flytestdlib#50
* lyft/flyteidl#27 (depends on lyft/flytestdlib#50 to be merged and released)

### CODE REVIEW

File: boilerplate/update.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ set -e
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
 
 OUT="$(mktemp -d)"
-git clone git@github.com:lyft/boilerplate.git "${OUT}"
+git clone --single-branch --branch gomod git@github.com:lyft/boilerplate.git "${OUT}"
```
Comment (wild-endeavor) [inline boilerplate/update.sh]: will revert this before merging, just here for testing.

File: boilerplate/lyft/golang_test_targets/Makefile [hunk 2]
```diff
@@ -1,25 +1,30 @@
...
+#
 # TO OPT OUT OF UPDATES, SEE https://github.com/lyft/boilerplate/blob/master/Readme.rst
 
-DEP_SHA=1f7c19e5f52f49ffb9f956f64c010be14683468b
 
 .PHONY: lint
 lint: #lints the package for common code smells
-	which golangci-lint || curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | bash -s -- -b $$GOPATH/bin v1.16.0
-	golangci-lint run --exclude deprecated
+	which golangci-lint || make download_tooling
```
[the remark this answers is not in the document]
Comment (honnix) [reply boilerplate/lyft/golang_test_targets/Makefile]: `lint: download_tooling`

File: boilerplate/lyft/golang_test_targets/download_tooling.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+# Everything in this file needs to be installed outside of current module
+# The reason we cannot turn off module entirely and install is that we need the replace statement in go.mod
+# because we are installing a mockery fork. Turning it off would result installing the original not the fork.
+# However, because the installation of these tools themselves sometimes modifies the go.mod/go.sum files. We don't
+# want this either. So instead, we're going to copy those files into a temporary directory, do the installation, and=
```
Comment (honnix) [inline boilerplate/lyft/golang_test_targets/download_tooling.sh]: typo `and=`

File: boilerplate/lyft/golang_test_targets/download_tooling.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+tools=(
+  "github.com/vektra/mockery/cmd/mockery"
+  "github.com/lyft/flytestdlib/cli/pflags"
+  "github.com/golangci/golangci-lint/cmd/golangci-lint"
+  "github.com/alvaroloes/enumer"
+)
+
+for tool in "${tools[@]}"
+do
+    go_install_tool $tool
```
Comment (honnix) [inline boilerplate/lyft/golang_test_targets/download_tooling.sh]: Maybe do a `which ... || ...` here?

**Comment by honnix** [comment]
Tried addressing comments in #5

File: boilerplate/lyft/golang_test_targets/download_tooling.sh [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  cp go.mod go.sum "$tmp_dir"
+  pushd "$tmp_dir"
+  go get "$1"
+  popd
+}
+
+# List of tools to go get
+# In the format of "<cli>:<package>" or ":<package>" if no cli
+tools=(
+  "mockery:github.com/vektra/mockery/cmd/mockery"
```
Comment (wild-endeavor) [inline boilerplate/lyft/golang_test_targets/download_tooling.sh]: @honnix let me take one more look tomorrow before merging this PR.  I want to add version numbers in here as well.

File: boilerplate/lyft/golang_test_targets/Makefile [hunk 6]
```diff
@@ -1,25 +1,29 @@
 # WARNING: THIS FILE IS MANAGED IN THE 'BOILERPLATE' REPO AND COPIED TO OTHER REPOSITORIES.
 # ONLY EDIT THIS FILE FROM WITHIN THE 'LYFT/BOILERPLATE' REPOSITORY:
-# 
+#
 # TO OPT OUT OF UPDATES, SEE https://github.com/lyft/boilerplate/blob/master/Readme.rst
 
-DEP_SHA=1f7c19e5f52f49ffb9f956f64c010be14683468b
+
```
Comment (honnix) [inline boilerplate/lyft/golang_test_targets/Makefile]: Oops it seems I introduced this empty line.

File: boilerplate/lyft/golang_test_targets/download_tooling.sh [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+echo "Using temp directory ${tmp_dir}"
+cp -R boilerplate/lyft/golang_support_tools/* $tmp_dir
+pushd "$tmp_dir"
+
+for tool in "${tools[@]}"
+do
+    cli=$(echo "$tool" | cut -d':' -f1)
+    package=$(echo "$tool" | cut -d':' -f2)
+    echo "Installing ${package}"
+    GO111MODULE=on go install "$package"
```
Comment (honnix) [inline boilerplate/lyft/golang_test_targets/download_tooling.sh]: So we install it anyway without checking existence? If that is the case we don't need that column hack.

**Comment by honnix** [comment]
Minor comment otherwise LGTM. 👍

File: boilerplate/lyft/golang_test_targets/download_tooling.sh [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+set -e
+
+# List of tools to go get
+# In the format of "<cli>:<package>" or ":<package>" if no cli
+tools=(
+  "mockery:github.com/vektra/mockery/cmd/mockery"
+  "pflags:github.com/lyft/flytestdlib/cli/pflags"
+  "golangci-lint:github.com/golangci/golangci-lint/cmd/golangci-lint"
+  "enumer:github.com/alvaroloes/enumer"
```
Comment (EngHabu) [inline boilerplate/lyft/golang_test_targets/download_tooling.sh]: Can you add this pleaaase

File: boilerplate/lyft/golang_test_targets/download_tooling.sh (hunk 7 above)
Comment (wild-endeavor) [reply boilerplate/lyft/golang_test_targets/download_tooling.sh]: yeah, cuz we want to make sure the version is the right one

Comment (honnix) [reply boilerplate/lyft/golang_test_targets/download_tooling.sh]: Yeh I see that point totally. We can remove that column cut thing so the code became much simpler. Also I would suggest we enable shellcheck on TravisCI.

**Comment by honnix** [comment]
👏

**Pull request merged**